beacon-chain/rpc/prysm/v1alpha1/validator/proposer_execution_payload.go

```diff
@@ -147,7 +148,19 @@ func (vs *Server) getExecutionPayload(ctx context.Context, slot types.Slot, vIdx
 	if payloadID == nil {
 		return nil, errors.New("nil payload id")
 	}
-	return vs.ExecutionEngineCaller.GetPayload(ctx, *payloadID)
+	payload, err := vs.ExecutionEngineCaller.GetPayload(ctx, *payloadID)
+	if err != nil {
+		return nil, err
+	}
+	// Warn if the fee recipient is not the value we expect.
+	if payload != nil && !bytes.Equal(payload.FeeRecipient, feeRecipient[:]) {
+		logrus.WithFields(logrus.Fields{
+			"wantedFeeRecipient": fmt.Sprintf("%#x", feeRecipient),
+			"received":           fmt.Sprintf("%#x", payload.FeeRecipient),
+		}).Warn("Fee recipient address from execution client is not what was expected. " +
```

Comment:
I think we should add some actionable tasks here. maybe like " please verify the execution client only received 1 fee recipient setting."